Services/Mail/classes/Preview/class.ilMailPreviewGUI.php

```diff
@@ -0,0 +1,87 @@
+<?php
+/**
+ * Copyright (c) 2017 ILIAS open source, Extended GPL, see docs/LICENSE
+ * CaT Concepts and Training GmbH
+ */
+
+/**
+ * Preview form for mail templates
+ *
+ * @author Stefan Hecken 	<stefan.hecken@concepts-and-training.de>
+ */
+class ilMailPreviewGUI {
+	const PREVIEW_CLASS_PREVIX = "Preview";
+	/**
+	 * @var ilMailTemplate
+	 */
+	protected $template_id;
+
+	public function __construct(ilMailTemplate $template) {
+		global $DIC;
+		$this->g_user = $DIC->user();
+		$this->g_lng = $DIC->language();
+		$this->template = $template;
+	}
+
+	/**
+	 * Render mail preview
+	 *
+	 * @return string
+	 */
+	public function getHTML() {
+		require_once 'Services/Form/classes/class.ilPropertyFormGUI.php';
+		$form = new ilPropertyFormGUI();
+		$form->setPreventDoubleSubmission(false);
+		$form->setTableWidth('100%');
+		$form->setTitle("Vorschau für: ".$this->template->getTitle());
+
+		require_once("Services/Mail/classes/class.ilMail.php");
+		$from = new ilCustomInputGUI($this->g_lng->txt('from'));
+		$from->setHtml(ilUtil::img(ilUtil::getImagePath('HeaderIconAvatar.svg'), ilMail::_getIliasMailerName()) . '<br />' . ilMail::_getIliasMailerName());
+		$form->addItem($from);
+
+		$to = new ilCustomInputGUI($this->g_lng->txt('mail_to'));
+		$to->setHtml(ilUtil::htmlencodePlainString($this->g_lng->txt('user'), false));
+		$form->addItem($to);
+
+		$subject = new ilCustomInputGUI($this->g_lng->txt('subject'));
+		$subject->setHtml(ilUtil::htmlencodePlainString($this->template->getSubject(), true));
+		$form->addItem($subject);
+
+		$message = new ilCustomInputGUI($this->g_lng->txt('message'));
+		$message->setHtml($this->populatePlaceholder($this->template->getMessage()));
+		$form->addItem($message);
+
+		return $form->getHtml();
+	}
+
+	/**
+	 * Replace placeholders with default values
+	 *
+	 * @param string 	$message
+	 *
+	 * @return string
+	 */
+	protected function populatePlaceholder($message) {
+		$context_class_preview = $this->getPreviewClassNameFromContext();
+		require_once "Services/Mail/classes/Preview/class.".$context_class_preview.".php";
```

Comment:
Was, wenn die Preview-Klasse nicht existiert?

Reply:
Wenn die Preview-Klasse nicht existiert kommt hier eine FileNotFound Fehlermeldung.

Reply:
Hab ich mir fast gedacht ;)
Kann ja auch nicht wirklich, wenn du in class.ilMailTemplateTableGUI.php auf "instanceof ilCourseMailTemplateTutorContext" prüfst.
Alternativ könntest du auch auf die Existenz der Datei checken (in ilMailTemplateTableGUI). Nur so ne Idee...